dali/pipeline/operators/decoder/nvjpeg_allocator.h

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATORS_DECODER_NVJPEG_ALLOCATOR_H_
+#define DALI_PIPELINE_OPERATORS_DECODER_NVJPEG_ALLOCATOR_H_
+
+
+#include <cuda_runtime_api.h>
+
+#include <memory>
+#include <mutex>
+#include <numeric>
+#include <string>
+#include <utility>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include "dali/common.h"
+#include "dali/error_handling.h"
+#include "dali/util/cuda_utils.h"
+
+namespace dali {
+
+namespace memory {
+
+using PinnedMemoryPtr = void*;
+
+class BasicPinnedAllocator {
+ public:
```

Comment:
I didn't spotted any usage of BasicPinnedAllocator. Is it intentionally retained unused?

Reply:
Not decided yet what to do about it